File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        prev = nums[0]
        res = max(float(-inf),prev)
        for i in range(1,len(nums)):
            curr = nums[i]
            if curr > prev + curr:
                prev = curr
            else:
                prev = prev + curr
            res = max(res,prev)
        return res

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int res = Integer.MIN_VALUE;
        int sum = nums[0];
        for(int i = 1;i<nums.length;i++){
            res = Math.max(res,sum);
            if(sum + nums[i] < nums[i]){
                sum = nums[i];
            }
            else{
                sum = sum + nums[i];
            }
        } 
        return Math.max(res,sum);
    }
}